apps/masterbots.ai/components/routes/chat/chat.tsx

```
@@ -113,18 +117,42 @@ export function Chat({
+        setActiveTool(toolCall.toolName as WordWareFlowPaths);
       },
-      onError(error) {
+      async onError(error) {
         console.error("Error in chat: ", error);
         toast.error("Failed to send message. Please try again.");
+
+        if (isNewChat) {
+          await deleteThread({
+            threadId,
+            jwt: session!.user?.hasuraJwt,
+            userId: session!.user.id,
```

Biome

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)